include/igl/writeMSH.cpp

```diff
@@ -0,0 +1,159 @@
+/* high level interface for MshSaver*/
+
+/* Copyright (C) 2020 Vladimir Fonov <vladimir.fonov@gmail.com> */
+/*
+/* This Source Code Form is subject to the terms of the Mozilla */
+/* Public License v. 2.0. If a copy of the MPL was not distributed */
+/* with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "writeMSH.h"
+#include "MshSaver.h"
+#include "MshLoader.h"
+#include <iostream>
+
+
+#if __cplusplus==201103L
+namespace igl {
+    // helper function, unfortunately impossible to make lambda in c++11
+    template <typename T,typename Derived> 
+    void __append_mat_to_vec(std::vector<T> &vec, const Eigen::PlainObjectBase<Derived> & mat) 
```

Comment:
I think double underscore names are reserved for compiler's internal usage.  This helper function can be declared in anonymous namespace to avoid symbol leaking.

Reply:
ok, using anonymous namespace.

Reply:
hmm since libigl is a header-only library, maybe using a `namespace internal` might be preferable here.

Reply:
Good point, perhaps `internal` namespace is more preferable in this case.  Sorry.

Reply:
so, 
```
namespace igl {
 namespace {
   template <typename T,typename Derived> 
    void __append_mat_to_vec(...
...
```
?

Reply:
```cpp
namespace igl {
namespace internal {

template<...>
void append_mat_to_vec() {
...
}

}
}
```